File: 0268_missing_number.swift
// https://leetcode.com/problems/missing-number/

class Solution {
    func missingNumber(_ nums: [Int]) -> Int {
        let set = Set(nums)
        for i in 0...nums.count {
            if !set.contains(i) {
                return i
            }
        }
        return 0
    }
}

File: 0268_missing_number.kt
// https://leetcode.com/problems/missing-number/description/

class Solution {
    fun missingNumber(nums: IntArray): Int {
        val set = nums.toSet()
        var i = 0
        while (true) {
            if (set.contains(i)) {
                i++
            } else {
                return i
            }
        }
    }
}
